web/source/kmwcallback.js

```diff
@@ -290,13 +290,21 @@ if(!window['tavultesoft']['keymanweb']['initialized']) {
       var retVal = 0; // I3318
       var keyCode = (e.Lcode == 173 ? 189 : e.Lcode);  //I3555 (Firefox hyphen issue)
 
+      var bitmask;
+      if(keymanweb._ActiveKeyboard['KV']) {
```

Comment:
I reduced the EuroLatin2 keyboard to these 2-lines
```
store(controls) "@"
+ any(controls) > index(controls, 1) dk(1)
```
Compiles to this
```javascript
        if (k.KKM(e, 16400, 50)) {
            r = m = 1;
            k.KO(0, t, "@");
            k.KDO(-1, t, 0);
        }
```

Typing a `2` drops into this statement and outputs `@`. Maybe something in keymanweb.KKM is incomplete for deadkeys?

Reply:
I'm wondering if you need to initialize
`var bitmask = osk.modifierBitmasks.NON_CHIRAL;`

That seems to fix the comparison `Lruleshift & bitmask` and makes EuroLatin2 work for me.